psi4/src/psi4/libsapt_solver/exch-ind30.cc

```diff
@@ -314,5 +318,258 @@ double SAPT2p3::exch_ind30_3(double **sBS) {
 
     return (energy);
 }
+
+// Compute the nonapproximated third-order exchange-induction //
+// Konrad Patkowski, based on Jonathan Waldrop's Psi4NumPy code //
+// September 2021 //
+void SAPT2p3::sinf_e30ind() {
+
+    if (print_) {
+        outfile->Printf("  ==> Nonapproximated third-order induction <==\n\n");
+    }
+
+    // => Sizing <= //
+
+    int nn = basisset_->nbf();
+    int na = noccA_;
+    int nb = noccB_;
+    int nr = nvirA_;
+    int ns = nvirB_;
+
+    auto uAR = Matrix("Ind30 uAR Amplitudes", na, nr);
+    auto uBS = Matrix("Ind30 uBS Amplitudes", nb, ns);
+    uAR.load(psio_, PSIF_SAPT_AMPS, Matrix::SaveType::Full);
+    uBS.load(psio_, PSIF_SAPT_AMPS, Matrix::SaveType::Full);
+
+    // => Intermolecular overlap matrix and inverse <= //
+    auto Sab = linalg::triplet(CoccA_, Smat_, CoccB_, true, false, false);
+
+    double** Sabp = Sab->pointer();
+    auto D = Matrix("D", na + nb, na + nb);
+    D.identity();
+    double** Dp = D.pointer();
+    for (int a = 0; a < na; a++) {
+        for (int b = 0; b < nb; b++) {
+            Dp[a][b + na] = Dp[b + na][a] = Sabp[a][b];
+        }
+    }
+    D.power(-1.0, 1.0E-12);
```

Comment:
There's also a `Matrix.general_inverse` that may be of interest: https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libmints/matrix.h#L959-L992